include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge.h

```diff
@@ -391,7 +415,7 @@ __parallel_merge(oneapi::dpl::__internal::__device_backend_tag, _ExecutionPolicy
 
     using __value_type = oneapi::dpl::__internal::__value_t<_Range3>;
 
-    const std::size_t __n = __rng1.size() + __rng2.size();
+    const std::uint64_t __n = std::min<std::uint64_t>(__rng1.size() + __rng2.size(), __rng3.size());
```

Comment:
Why we can't use `std::size_t` here as were before?

Reply:
ok, lets try